File: samples/FsBulletML.Sample/Background.fs
namespace FsBulletML.Sample

open System
open Microsoft.Xna.Framework
open Microsoft.Xna.Framework.Graphics

type Background (texture, screenHeight, scrollSpeed) as this =
  [<DefaultValue>]val mutable private texture : Texture2D 
  [<DefaultValue>]val mutable private pos : Vector2 
  [<DefaultValue>]val mutable private originPos : Vector2 
  [<DefaultValue>]val mutable private scrollSpeed : float32
  [<DefaultValue>]val mutable private screenHeight : float32

  do
    this.scrollSpeed <- scrollSpeed
    this.screenHeight <- screenHeight
    this.texture <- texture
    this.originPos <- new Vector2(0.f, 0.f)
    this.pos <- new Vector2(this.originPos.X, this.originPos.Y)

  member this.Update(delta:float) =
    this.pos.Y <- this.pos.Y + this.scrollSpeed * (delta |> float32)
    this.pos.Y <- this.pos.Y % (float32 this.texture.Height)

  member this.Draw(sp:SpriteBatch) =
    if (this.pos.Y < this.screenHeight) then
      sp.Draw(this.texture, this.pos, Color.White)
    let drawPos = this.pos - new Vector2(0.f, this.texture.Height |> float32)
    sp.Draw(this.texture, drawPos, Color.White)

File: samples/FsBulletML.Sample.MonoGame.CSharp/Background.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;

namespace FsBulletML.Sample.MonoGame.CSharp
{
    public class Background
    {
        Texture2D texture; 
        Vector2 position;
        Vector2 textureOrigin;
        float scrollSpeed;
        int screenHeight;

        public Background(Texture2D texture, int screenHeight, float scrollSpeed)
        {
            this.scrollSpeed = scrollSpeed;
            this.screenHeight = screenHeight;
            this.texture = texture;
            this.textureOrigin = new Vector2(texture.Width / 2, 0);
            this.position = new Vector2(textureOrigin.X, screenHeight / 2);
        }

        public void Update(float delta)
        {
            this.position.Y += this.scrollSpeed * delta;
            this.position.Y = this.position.Y % this.texture.Height;
        }

        public void Draw(SpriteBatch sp)
        {
            if (this.position.Y < this.screenHeight)
                sp.Draw(this.texture, this.position, null, Color.White, 0.0f, this.textureOrigin, 1.0f, SpriteEffects.None, 0.0f);
            sp.Draw(this.texture, this.position - new Vector2(0, this.texture.Height), null, Color.White, 0.0f, this.textureOrigin, 1.0f, SpriteEffects.None, 0.0f);
        }
    }
}
